code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -42,13 +64,19 @@ var/list/gaslist_cache = null
 
 /datum/gas_mixture/New(volume = CELL_VOLUME)
 	..()
-	gases = new
+	gases = INIT_GASES
 	temperature = 0
 	temperature_archived = 0
 	src.volume = volume
 	last_share = 0
 	fuel_burnt = 0
 
+/datum/gas_mixture/Destroy()
+	gases = null
+	holder = null
+	. = QDEL_HINT_PUTINPOOL

```

Comment:
I'm gonna tell you right now that the pooling is gonna be likely to be causing an overhead that wouldn't exist without it, both on qdel, and on fetching a item out of the pool.

byond already does pooling internally, and the cost of creating new objects is in the New(), not in creating the actual object, something the pooling doesn't properly address.

I've wanted to believe, but pooling is currently more costly because it has to call New() still, and object creation overhead is rarely above just the overhead of the proc call to poolornew.

This change will most likely have to go. I'll see about profiling what this change added if it's easy to revert it


As an aside, atmos doesn't use qdel because it's a closed system, we don't do checks in atmos when we don't have to, this includes checking for objects leaking (all that qdel basically does) this overhead is likely to not be worth it.

atm a gas_mixture should never be needed to be explicitly deleted, if something is referencing it, it needs to exist, so it's better to just let byond's GC handling work this out and save the overhead in temp gas mixtures. 
If a gas mixture is ever needed to be hard deleted, it is more likely the bug is in the fact it was ever sent to qdel(ie something was being too eager to qdel references on it), an issue that doesn't exist if you stick to byond's built-in referencing counting stuff.

Reply:
you were right